lib/phoenix_component.ex

```diff
@@ -385,43 +545,62 @@ defmodule Phoenix.Component do
     })
   end
 
-  @builtin_types [:boolean, :integer, :float, :string, :atom, :list, :map]
+  @builtin_types [:boolean, :integer, :float, :string, :atom, :list, :map, :global]
   @valid_types [:any] ++ @builtin_types
 
-  defp validate_attr_type!(name, type, line, file) when is_atom(type) do
+  defp validate_attr_type!(module, name, type, line, file) when is_atom(type) do
+    attrs = get_attrs(module)
+
+    cond do
+      Enum.find(attrs, fn attr -> attr.name == name end) ->
+        compile_error!(line, file, """
+        a duplicate attribute with name #{inspect(name)} already exists\
```

Comment:
```suggestion
        a duplicate attribute with name #{inspect(name)} already exists
```

I guess that's not intentional

Reply:
José doesn't like unnecessary newlines, so intentional in this case :D